gmail/src/views/login.ts

```diff
@@ -0,0 +1,75 @@
+import { formatUrl, repeat } from "../utils/format";
+import { notify, createKeyValueWidget } from "./helpers";
+import { State } from "../models/state";
+import { IMAGES_LOGIN } from "./icons";
+
+function onNextLogin(event) {
+    const validatedUrl = formatUrl(event.formInput.odooServerUrl);
+
+    if (!validatedUrl) {
+        return notify("Invalid URL");
+    }
+
+    State.odooServerUrl = validatedUrl;
+
+    return CardService.newActionResponseBuilder()
+        .setOpenLink(
+            CardService.newOpenLink()
+                .setUrl(State.odooLoginUrl)
+                .setOpenAs(CardService.OpenAs.OVERLAY)
+                .setOnClose(CardService.OnClose.RELOAD_ADD_ON),
+        )
+        .build();
+}
+
+export function buildLoginMainView() {
+    const card = CardService.newCardBuilder();
+
+    // Trick to make large centered button
+    const invisibleChar = "⠀";
+
+    card.addSection(
+        CardService.newCardSection()
+            .addWidget(
+                CardService.newImage().setAltText("Connect to your Odoo database").setImageUrl(IMAGES_LOGIN.main_image),
+            )
+            .addWidget(
+                CardService.newTextInput()
+                    .setFieldName("odooServerUrl")
+                    .setTitle("Database URL")
+                    .setHint("e.g. company.odoo.com")
+                    .setValue(State.odooServerUrl || ""),
+            )
+            .addWidget(
+                CardService.newTextButton()
+                    .setText(repeat(invisibleChar, 12) + "Login" + repeat(invisibleChar, 12))
```

Comment:
So that's where you use that repeat thing... waw ^^

Reply:
Yes xD to make a "large" button, because I can not change the width ^^ (and we want a beautifoul login page)